File: backend/src/main/java/com/bluefields/transporte/domain/entity/Bus.java
package com.bluefields.transporte.domain.entity;

import jakarta.persistence.*;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "bus")
public class Bus {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "empresa_id", nullable = false)
    private Empresa empresa;

    @Column(name = "numero_interno", nullable = false, length = 20)
    private String numeroInterno;

    @Column(nullable = false, length = 20)
    private String placa;

    @Column(nullable = false)
    private Integer capacidad;

    @Column(nullable = false)
    private Integer filas;

    @Column(nullable = false)
    private Boolean activo = true;

    @Column(name = "foto_url", length = 500)
    private String fotoUrl;

    /** Terminal donde opera el bus (Bluefields o Managua). */
    @Column(nullable = false, length = 100)
    private String sede = "Bluefields";

    @OneToMany(mappedBy = "bus", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("numero ASC")
    private List<AsientoBus> asientos = new ArrayList<>();

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    void prePersist() {
        Instant now = Instant.now();
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    void preUpdate() {
        updatedAt = Instant.now();
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Empresa getEmpresa() { return empresa; }
    public void setEmpresa(Empresa empresa) { this.empresa = empresa; }
    public String getNumeroInterno() { return numeroInterno; }
    public void setNumeroInterno(String numeroInterno) { this.numeroInterno = numeroInterno; }
    public String getPlaca() { return placa; }
    public void setPlaca(String placa) { this.placa = placa; }
    public Integer getCapacidad() { return capacidad; }
    public void setCapacidad(Integer capacidad) { this.capacidad = capacidad; }
    public Integer getFilas() { return filas; }
    public void setFilas(Integer filas) { this.filas = filas; }
    public Boolean getActivo() { return activo; }
    public void setActivo(Boolean activo) { this.activo = activo; }
    public String getFotoUrl() { return fotoUrl; }
    public void setFotoUrl(String fotoUrl) { this.fotoUrl = fotoUrl; }
    public String getSede() { return sede; }
    public void setSede(String sede) { this.sede = sede; }
    public List<AsientoBus> getAsientos() { return asientos; }
    public void setAsientos(List<AsientoBus> asientos) { this.asientos = asientos; }
    public Instant getCreatedAt() { return createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
}

File: mobile/lib/core/models/bus.dart
class AsientoBus {
  AsientoBus({
    required this.id,
    required this.numero,
    required this.fila,
    required this.posicion,
  });

  final int id;
  final int numero;
  final int fila;
  final String posicion;

  factory AsientoBus.fromJson(Map<String, dynamic> json) => AsientoBus(
        id: json['id'] as int,
        numero: json['numero'] as int,
        fila: json['fila'] as int,
        posicion: json['posicion'] as String,
      );
}

class Bus {
  Bus({
    required this.id,
    required this.empresaId,
    required this.numeroInterno,
    required this.placa,
    required this.capacidad,
    required this.filas,
    required this.activo,
    this.fotoUrl,
    required this.sede,
    this.asientos = const [],
  });

  final int id;
  final int empresaId;
  final String numeroInterno;
  final String placa;
  final int capacidad;
  final int filas;
  final bool activo;
  final String? fotoUrl;
  final String sede;
  final List<AsientoBus> asientos;

  factory Bus.fromJson(Map<String, dynamic> json) {
    final asientosJson = json['asientos'] as List<dynamic>? ?? [];
    return Bus(
      id: json['id'] as int,
      empresaId: json['empresaId'] as int,
      numeroInterno: json['numeroInterno'] as String,
      placa: json['placa'] as String,
      capacidad: (json['capacidad'] as num).toInt(),
      filas: (json['filas'] as num?)?.toInt() ?? 0,
      activo: json['activo'] as bool? ?? true,
      fotoUrl: json['fotoUrl'] as String?,
      sede: json['sede'] as String,
      asientos: asientosJson.map((e) => AsientoBus.fromJson(e as Map<String, dynamic>)).toList(),
    );
  }
}
